File: javascript/leetcode/pascals-triangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    iterations = 0
    result = []
    while (iterations < numRows) {
        result.push(generateRow(result[iterations - 1]))
        iterations++
    }
    return result
};

var generateRow = function(lastRow) {
    if (!lastRow) {return [1]}
    newRow = []
    newRow.push(lastRow[0])
    for (var i = 0; i < lastRow.length - 1; i++) {
        newRow.push(lastRow[i] + lastRow[i + 1])
    }
    newRow.push(lastRow[lastRow.length -1])
    return newRow
}

File: python/leetcode/pascals-triangle.py
class Solution:
    def generate(self, num_rows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        if num_rows == 0:
            return []
        iterations = 1
        result = [[1]]
        while iterations < num_rows:
            result.append(self.generate_row(result[iterations - 1]))
            iterations += 1
        return result

    def generate_row(self, last_row):
        new_row = []
        new_row.append(last_row[0])
        for i in range(len(last_row) - 1):
            new_row.append(last_row[i] + last_row[i + 1])
        new_row.append(last_row[-1])
        return new_row
